test/app/exercises_test.rb

```diff
@@ -9,11 +9,27 @@ def app
     ExercismWeb::App
   end
 
-  def test_excercises_by_key
-    alice = User.create(username: 'alice', github_id: 1)
-    exercise = UserExercise.create(user: alice, language: 'go', slug: 'one')
-    submission = Submission.create(user: alice, language: 'go', slug: 'bob', user_exercise: exercise)
+  attr_reader :alice, :exercise, :submission
+
+  def setup
+    super
+    @alice = User.create(username: 'alice', github_id: 1)
+    @exercise = UserExercise.create(user: alice, language: 'go', slug: 'one')
+    @submission = Submission.create(user: alice, language: 'go',
+                                    slug: 'bob', user_exercise: exercise)
+  end
+
+  def test_exercises_next_with_session_values
+    get "/exercises/next", {}, login(alice)
+    assert_equal 302, last_response.status
+  end
 
+  def test_exercises_next_without_session_values
+    get "/exercises/next"
+    assert_equal 302, last_response.status
```

Comment:
If there's no user, then it will redirect because of the _please login_ behavior.